.github/workflows/build.yml

```diff
@@ -145,9 +148,9 @@ jobs:
         --build-arg BUILDKIT_CONTEXT_KEEP_GIT_DIR=true 
         https://github.com/stellar/go.git#${{ env.GO_REPO_BRANCH }}
     - name: Upload Stellar-Friendbot Image
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@v4
       with:
-        name: image-stellar-friendbot-${{ inputs.arch }}
+        name: image-stellar-friendbot-${{ inputs.tag }}-${{ inputs.arch }}
```

Comment:
for the artifact upload/download naming, does it need to include the tag in postfix, within each workflow, there is only one tag name, but arch would change twice, etc. wasn't sure if this was related to preventing [v4 breaking change on same artifact name](https://github.com/actions/upload-artifact?tab=readme-ov-file#not-uploading-to-the-same-artifact)

Reply:
Could you elaborate on the concern? The tag is included in the name to ensure artifacts are unique for the new single large workflow. Prior to this change the latest, testing, and future variants ran in their own top level workflows which meant that their artifacts only had to be unique within their own workflows. Now taht they run in a single shared workflow their artifacts need to be unique across them all.

Reply:
ah, ok, i didn't pick up on subtlety of that workflow scope change during variant builds, cool, makes sense now, thanks for clarification